pkg/storage/chunked.go

```diff
@@ -318,8 +330,22 @@ func (c *chunkedSecrets) Query(queryLabels map[string]string) ([]*release.Releas
 	c.Log("query: labels=%v", queryLabels)
 	defer c.Log("queried: labels=%v", queryLabels)
 
+	// The only labels that get stored on the index secret are system labels, so we'll do a two-pass
+	// query. First, we'll request index secrets from the API server that match the query labels that
+	// are system labels. From there, we decode the releases that match, and then further filter those
+	// based on the rest of the query labels that are not system labels.
```

Comment:
This was necessary to get existing tests to pass. We have a test that uses a custom label in a query.